api/agentpb/agent.proto

```diff
@@ -391,6 +391,27 @@ message PBMSwitchPITRResponse {
   // Error message.
   string error = 1;
 }
+// ParseDefaultsFileRequest is an ServerMessage asking pmm-agent to parse defaults file.
+message ParseDefaultsFileRequest {
+  // Service type.
+  inventory.ServiceType service_type = 1;
+
+  // Defaults config path
+  string config_path = 2;
+}
+
+// ParseDefaultsFileResponse is an AgentMessage containing a result of parding defaults file.
+message ParseDefaultsFileResponse {
+
+  // Error message if parse failed.
+  string error = 1;
+
+  string username = 2;
+  string password = 3;
+  string host = 4;
+  uint32 port = 5;
+
```

Comment:
my.cnf might contain path to socket, so we should return it too

Reply:
Added.